### PR TITLE
Add Documentation for Packaging Applications

### PR DESCRIPTION
**Pull request by PIYUSHgoyal16**
Add Documentation for Packaging Applications and Releasing them on Snap Store

### CODE REVIEW

File: _documentation/04-packaging-drivers.md [hunk 1]
```diff
@@ -1,4 +1,295 @@
...
+
+* Snaps are easily discoverable and installable from the Snap Store, an app store with an audience of millions who can browse and install snaps graphically in the Snap Store or from the command-line.
+
+You are encouraged to dive deep into the official <a href="https://snapcraft.io/docs">Snap Documentation</a> to explore more about it.
+
+<h2 id="package"> Packaging Drivers </h2>
+
+Most Snaps we create are driven using `snapcraft.yaml`. This file describes a snap’s build dependencies and run-time requirements, it integrates remote repositories and extensions and runs custom scripts and hooks for better integration with CI systems.
+
+As a distributor, you would create a `snapcraft.yaml` file and upload the same on Snap Store. When the user runs `snap install`, snapcraft will fetch and open the corresponding `snapcraft.yaml`, interpret it, figure out what to do, and build your application.
```
Comment (tillkamppeter) [inline _documentation/04-packaging-drivers.md:37]: As the developer you will not only upload the snapcraft.yaml, but also auxiliary files. See the cups-snap repository on OpenPrinting. All files of it get uploaded. Same also for the hp-printer-app respository. The Snap Store builds the Snap for several architectures (currently 6 AFAIR) and keep them on their servers for download. If a user runs "snap install ..." the package gets downloaded to the user. Note that the build only happens once with the developer's upload and not with each user's download (imaging what would happen with LibreOffice if the Snap Store would build it on each download request).

File: _documentation/04-packaging-drivers.md [hunk 2]
```diff
@@ -1,4 +1,295 @@
...
+    
+    * **classic**: Allows access to your system’s resources in much the same way traditional packages do. To safeguard against abuse, publishing a classic snap requires manual approval, and installation requires the --classic command-line argument.
+
+    We would be using the confinement value as strict.
+
+        confinement: strict
+
+___
+
+There are other metadata that could be used to provide more information to snapcraft besides the above-listed ones. One may refer to <a href="https://snapcraft.io/docs/snapcraft-top-level-metadata">Snapcraft top-level format documentation</a> to know about these.
```
Comment (tillkamppeter) [inline _documentation/04-packaging-drivers.md:129]: There is other metadata ... "data" is never Plural.

Comment (PIYUSHgoyal16) [reply _documentation/04-packaging-drivers.md:129]: Thanks, Till, corrected.

File: _documentation/04-packaging-drivers.md (hunk 1 above)
Comment (PIYUSHgoyal16) [reply _documentation/04-packaging-drivers.md]: @tillkamppeter I have corrected the mistake. Kindly review and merge the changes.

**Review by tillkamppeter — APPROVED** [review]

**Comment by tillkamppeter** [comment]
Thank you very much, it is merged now.

**Pull request merged**